### PR TITLE
[12.0][ADD] product_pricelist_supplierinfo: Partner to get pricelist price

### PR DESCRIPTION
**Pull request by CarmenMiranda**
Adding the option to send a `partner_id` to get the pricelist price of a product and obtaining the `partner_id` of the customer to whom the price is being computed.

### CODE REVIEW

**Review by nhomar — APPROVED** [review]

**Comment by pedrobaeza** [comment]
There are several opened PRs for this module currently being discussed. Please check them before for harmonizing all of them.

File: product_pricelist_supplierinfo/models/product_template.py [hunk 1]
```diff
@@ -9,8 +9,8 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    def _get_supplierinfo_pricelist_price(
-            self, rule, date=None, quantity=None, product_id=None):
+    def _get_supplierinfo_domain(
+            self, rule, date=None, quantity=None, product_id=None, partner_id=False):
```
Comment (moylop260) [inline product_pricelist_supplierinfo/models/product_template.py:13]: Where are you using the new parameter `partner_id`?

**Comment by CarmenMiranda** [comment]
> There are several opened PRs for this module currently being discussed. Please check them before for harmonizing all of them.

Hi @pedrobaeza Looks like [PR!668](https://github.com/OCA/product-attribute/pull/667) will be merged, Should I work over the changes of that branch? 
Because there are some changes here and in another [PR!683](https://github.com/OCA/product-attribute/pull/683) that I'm working on, which also be affected by changes in that PR

**Comment by pedrobaeza** [comment]
Yeah, that's the idea. Can you please check that the PR works also for you and review it?

File: product_pricelist_supplierinfo/models/product_template.py (hunk 1 above)
Comment (CarmenMiranda) [reply product_pricelist_supplierinfo/models/product_template.py:13]: Hi @moylop260 After the merge of the PR 668, I changed the way I was using the new parameter, and now I'm using it on the method `_select_seller`

**Review by moylop260 — APPROVED** [review]
👍

**Comment by moylop260** [comment]
@CarmenMiranda 

Could you check the following way, please?
 - https://github.com/OCA/product-attribute/pull/694/files

**Comment by moylop260** [comment]
@eLBati @pedrobaeza 

Could you check both, please?
 - https://github.com/OCA/product-attribute/pull/668

**Comment by eLBati** [comment]
Does this overlap with https://github.com/OCA/product-attribute/pull/694 ?

What is the use case of this?

**Comment by CarmenMiranda** [comment]
> Does this overlap with #694 ?
> 
> What is the use case of this?

Hi @eLBati n this PR I'm using the partner that comes from the compute of the price rule, which can be a specific one, and it can be sent with context when you get the product [price](https://github.com/odoo/odoo/blob/12.0/addons/product/models/product.py#L174). For example, when you come from a sale the partner of the sale is sent by context and we will get only the seller for the sale partner.

**Comment by CarmenMiranda** [comment]
Hi @moylop260 @eLBati @pedrobaeza After more functional reviews, these changes were not the best for what was required, so I will close this PR

**Pull request closed without merging**